### PR TITLE
Clarify the copy for Mentions in notifications.

### LINKED ISSUE #2397: "$foo mentioned you" notifications generally lie.
### Steps to reproduce

1. You receive a notification saying "$foo mentioned you: $msg"
2. You wince because typically $foo didn't mention you. instead they either:
 * replied to a message you sent
 * used `@room`
 * or otherwise didn't actually intentionally mention you.

#### What did you expect?

imo, notifications shouldn't waste space with "$foo mentioned you" at all.  If we really want do to this, they certainly shouldn't include "$foo mentioned you" unless $foo really did intentionally mention you.

#### What happened instead?

All sorts of things seem to trigger this. I can't actually remember seeing this notification when someone actually intentionally mentioned me, in fact.

### Your phone model

_No response_

### Operating system version

_No response_

### Application version

508

### Homeserver

_No response_

### Will you send logs?

No

### PR DESCRIPTION
**Pull request by pixlwave**
Fixes #2397. Keeping this draft until it is confirmed to work on Android too (with the notification summaries).

Simulator Screenshot - iPhone 15 Pro - 2024-08-08 at 10 26 32

### CODE REVIEW

File: ElementX/Resources/Localizations/en.lproj/Localizable.strings [hunk 1]
```diff
@@ -283,6 +283,7 @@
 "notification_reaction_body" = "Reacted with %1$@";
 "notification_room_invite_body" = "Invited you to join the room";
 "notification_sender_me" = "Me";
+"notification_sender_mention_reply" = "%1$@ mentioned or replied";
```
Comment (pixlwave) [inline ElementX/Resources/Localizations/en.lproj/Localizable.strings:286]: Note to self: This string is hidden in Localazy until we've confirmed we're happy.

**Review by stefanceriu — APPROVED** [review]
looks sensible to me 👍

**Comment by pixlwave** [comment]
@jmartinesp Did you look into this on Android?

**Comment by jmartinesp** [comment]
> @jmartinesp Did you look into this on Android?

Not really, sorry. But I think it should be fairly easy to do what Amsha suggested and create a fake 'person' with the name `{person} mention or replied`.

**Comment by pixlwave** [comment]
No worries, in which case I'll rebase and merge this to close the iOS issue 👍

File: ElementX/Resources/Localizations/en.lproj/Localizable.strings (hunk 1 above)
Comment (pixlwave) [reply ElementX/Resources/Localizations/en.lproj/Localizable.strings]: I've made it public.

**Pull request merged**